src/utils/useMenuState.ts

```diff
@@ -0,0 +1,58 @@
+import { useState, useEffect } from 'react';
+import { Router } from 'next/router';
+
+interface MenuStateReturn {
+  isMenuOpen: boolean;
+  openMenu: (event?: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+  handleMenuClick: (
+    event?: React.MouseEvent<HTMLAnchorElement, MouseEvent>
+  ) => void;
+}
+
+/**
+ * Keep track of the menu state and handle the opening and closing.
+ * Returns controls to the state.
+ * @param defaultOpen
+ */
+export function useMenuState(defaultOpen = false): MenuStateReturn {
+  const [isMenuOpen, setMenu] = useState<boolean>(defaultOpen);
+
+  const openMenu = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent> | undefined
+  ): void => {
+    if (event) {
+      event.preventDefault();
+    }
+    setMenu(true);
+  };
+
+  const handleMenuClick = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent> | undefined
+  ) => {
+    if (event?.currentTarget) {
+      event.currentTarget.blur();
+    }
+  };
+
+  /**
+   * Handle the closing of the menu
+   * If closed with handleMenuClick the menu would open too soon
+   * This closes the menu at the moment the content is loaded and displayed,
+   * (just as non-JS would kept displayed)
+   * */
+  useEffect(() => {
+    const handleCloseMenu = () => {
+      setMenu(false);
+    };
+    Router.events.on('beforeHistoryChange', handleCloseMenu);
+    return () => {
+      Router.events.off('beforeHistoryChange', handleCloseMenu);
+    };
+  });
```

Comment:
It's a little bit code duplication but this is also an option IMO

```suggestion
  useEffect(() => {
    Router.events.on('beforeHistoryChange', () => setMenu(false));
    return () => {
      Router.events.off('beforeHistoryChange',  () => setMenu(false));
    };
  }, []);
```

Reply:
Not sure if these will do the same. For event handlers, especially event-un-handlers arrow functions do not work as arguments (only references to a named function / variable containing a function.)